sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_connection_async.py

```diff
@@ -58,24 +59,33 @@ class Connection(object):
     :param list(str) offered_capabilities: The extension capabilities the sender supports.
     :param list(str) desired_capabilities: The extension capabilities the sender may use if the receiver supports
     :param dict properties: Connection properties.
+    :keyword str transport_type: Determines if the transport type is Amqp or AmqpOverWebSocket.
+     Defaults to TransportType.Amqp. It will be AmqpOverWebSocket if using http_proxy.
+    :keyword Dict http_proxy: HTTP proxy settings. This must be a dictionary with the following
+     keys: `'proxy_hostname'` (str value) and `'proxy_port'` (int value). When using these settings,
+     the transport_type would be AmqpOverWebSocket.
+     Additionally the following keys may also be present: `'username', 'password'`.
     """
 
     def __init__(self, endpoint, **kwargs):
         parsed_url = urlparse(endpoint)
         self.hostname = parsed_url.hostname
+        self._transport_type = kwargs.pop('transport_type', TransportType.Amqp)
         if parsed_url.port:
             self.port = parsed_url.port
         elif parsed_url.scheme == 'amqps':
             self.port = SECURE_PORT
         else:
             self.port = PORT
         self.state = None
-
         transport = kwargs.get('transport')
         if transport:
             self.transport = transport
         elif 'sasl_credential' in kwargs:
-            self.transport = SASLTransport(
+            sasl_transport = SASLWithWebSocket if (
+                self._transport_type is TransportType.AmqpOverWebsocket or kwargs.get("http_proxy")
+                ) else SASLTransport
+            self.transport = sasl_transport(
```

Comment:
It looks like the `AsyncTransport` on line 94 does not take self._transport_type the way the sync Transport does. Do you know why this is? (or have links to docs that explain why :P)

Reply:
AsyncTransport works a little bit differently than the sync scenario - there is no Abstract Transport in async which is why we don't need to set which transport (among TCP, SSL) explicitly is my undesrtanding